src/ingredient/ingredient.controller.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+  Query,
+} from '@nestjs/common';
+import { IngredientService } from './ingredient.service';
+import { IngredientDto } from './dto';
+
+@Controller('ingredients')
+export class IngredientController {
+  constructor(private ingredientService: IngredientService) {}
+  @Post()
+  async createIngredient(@Body() dto: IngredientDto) {
+    return this.ingredientService.createIngredient(dto);
+  }
+
+  @Delete(':id')
+  async deleteIngredient(@Param('id') id: string) {
+    return this.ingredientService.deleteIngredient(id);
+  }
+
+  @Put(':id')
+  async updateIngredient(@Param() id: string, @Body() dto: IngredientDto) {
+    return this.ingredientService.updateIngredient(id, dto);
+  }
+
+  @Get()
+  async getIngredients(@Query('ids') ids?: string) {
```

Comment:
Let's create GetIngredientsQueryDto, where you would add validator for array of string and pass it on to this method

Reply:
also, in service, if array is empty, do not pass to where condition
id: {
in: ids?.length ? ids : undefined 
}